iceoryx_posh/test/CMakeLists.txt

```diff
@@ -64,6 +64,7 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
     set(TEST_CXX_FLAGS ${ICEORYX_WARNINGS})
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
     set(TEST_CXX_FLAGS PRIVATE ${ICEORYX_WARNINGS} ${ICEORYX_SANITIZER_FLAGS} -Wno-pedantic -Wno-conversion)
+    set(TEST_LINKER_FLAGS -Wl,-z,stack-size=5000000)
```

Comment:
Could you please introduce here as well a variable like `ICEORYX_POSH_TEST_STACK_SIZE` and use it here? Since we run those tests also on Windows and Mac OS what do you think about adding this stack adjustment also to those platforms with:
```
set(TEST_LINKER_FLAGS /STACK:3500000) // windows
```

Clang should be described here: https://stackoverflow.com/questions/18909395/how-do-i-increase-the-stack-size-when-compiling-with-clang-on-os-x